src/main/java/sirius/web/mails/MailSender.java

```diff
@@ -547,25 +553,35 @@ private void checkReplyTo() {
 
     private void sanitize() {
         if (Strings.isFilled(senderEmail)) {
-            senderEmail = senderEmail.replaceAll("\\s", "");
+            senderEmail = encodeIdnMailAddressToAsciiIfNecessary(senderEmail);
         }
         if (Strings.isFilled(senderName)) {
             senderName = senderName.trim();
         }
         if (Strings.isFilled(receiverEmail)) {
-            receiverEmail = receiverEmail.replaceAll("\\s", "");
+            receiverEmail = encodeIdnMailAddressToAsciiIfNecessary(receiverEmail);
         }
         if (Strings.isFilled(receiverName)) {
             receiverName = receiverName.trim();
         }
         if (Strings.isFilled(replyToEmail)) {
-            replyToEmail = replyToEmail.replaceAll("\\s", "");
+            replyToEmail = encodeIdnMailAddressToAsciiIfNecessary(replyToEmail);
         }
         if (Strings.isFilled(replyToName)) {
             replyToName = replyToName.trim();
         }
     }
 
+    private String encodeIdnMailAddressToAsciiIfNecessary(String mailAddress) {
+        mailAddress = mailAddress.replaceAll("\\s", "");
+        if (UserContext.getSettings().get("mail.usePunycode").asBoolean()) {
+            String user = mailAddress.substring(0, mailAddress.indexOf("@"));
```

Comment:
```suggestion
            String preDomainPart = mailAddress.substring(0, mailAddress.indexOf("@") + 1);
```
Then there is no need to add the @ again